psi4/src/psi4/libmints/matrix.h

```diff
@@ -1084,9 +1058,85 @@ class PSI_API Matrix : public std::enable_shared_from_this<Matrix> {
      * @param theta - the angle (in radians) about which to rotate
      */
     void rotate_columns(int h, int i, int j, double theta);
-    friend class Vector;
+
+    PSI_DEPRECATED(
+        "Using `Matrix::matrix` instead of `detail::matrix` is deprecated, and in 1.4 it will "
+        "stop working")
+    static double** matrix(int nrow, int ncol) { return detail::matrix(nrow, ncol); }
+
+    PSI_DEPRECATED(
+        "Using `Matrix::free` instead of `detail::free` is deprecated, and in 1.4 it will "
+        "stop working")
+    static void free(double** Block) { detail::free(Block); }
+
+    PSI_DEPRECATED(
+        "Using `Matrix::create` instead of `create` is deprecated, and in 1.4 it will "
+        "stop working")
+    static SharedMatrix create(const std::string& name, const Dimension& rows, const Dimension& cols) {
+        return create(name, rows, cols);
+    }
+
+    PSI_DEPRECATED(
+        "Using `Matrix::horzcat` instead of `horzcat` is deprecated, and in 1.4 it will "
+        "stop working")
+    static SharedMatrix horzcat(const std::vector<SharedMatrix>& mats) { return horzcat(mats); }
+
+    PSI_DEPRECATED(
+        "Using `Matrix::vertcat` instead of `vertcat` is deprecated, and in 1.4 it will "
+        "stop working")
+    static SharedMatrix vertcat(const std::vector<SharedMatrix>& mats) { return vertcat(mats); }
+
+    PSI_DEPRECATED(
+        "Using `Matrix::doublet` instead of `doublet` is deprecated, and in 1.4 it will "
+        "stop working")
+    static SharedMatrix doublet(const SharedMatrix& A, const SharedMatrix& B, bool transA = false,
+                                bool transB = false) {
+        return doublet(A, B, transA, transB);
+    }
+
+    PSI_DEPRECATED(
+        "Using `Matrix::triplet` instead of `triplet` is deprecated, and in 1.4 it will "
+        "stop working")
+    static SharedMatrix triplet(const SharedMatrix& A, const SharedMatrix& B, const SharedMatrix& C,
+                                bool transA = false, bool transB = false, bool transC = false) {
+        return triplet(A, B, C, transA, transB, transC);
+    }
 };
 
-}  // namespace psi
+/**
+ * Convenient creation function return SharedMatrix
+ */
+SharedMatrix create(const std::string& name, const Dimension& rows, const Dimension& cols);
```

Comment:
Since `create` has been freed from both `Matrix` and `Vector` perhaps `create_matrix` and `create_vector` would increase readability. Right now, return type and # of arguments are the only way to tell.

Reply:
I've decided to remove it altogether. `create` was used in just a couple of places...